File: algorithms/warmup/java/Sherlock_and_Squares.java
import java.util.*;

class Sherlock_and_Squares {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();

        while(T>0) {
            int a = sc.nextInt();
            int b = sc.nextInt();

            System.out.println(Math.round(Math.floor(Math.sqrt(b)) - Math.ceil(Math.sqrt(a)) + 1));
            T--;
        }
        sc.close();
    }
}

File: algorithms/warmup/c/sherlock-and-squares.c
#include <stdio.h>
#include <math.h>

int main()
{
    int a, b;
    int T;

    scanf("%d", &T);

    while(T--){
    scanf("%d %d", &a, &b);
    printf("%d\n", (int)(floor(sqrt(b)) - ceil(sqrt(a)) + 1));
    }

    return 0;
}